src/avram/associations/has_many.cr

```diff
@@ -37,15 +37,16 @@ module Avram::Associations::HasMany
       type: {{ type_declaration.type }},
       foreign_key: :{{ foreign_key }},
       through: {{ through }},
-      relationship_type: :has_many
+      relationship_type: :has_many,
+      query: {{ query }}
 
     {% model = type_declaration.type %}
 
-    define_has_many_lazy_loading({{ assoc_name }}, {{ model }}, {{ foreign_key }}, {{ through }})
-    define_has_many_base_query({{ assoc_name }}, {{ model }}, {{ foreign_key }}, {{ through }})
+    define_has_many_lazy_loading({{ assoc_name }}, {{ model }}, {{ foreign_key }}, {{ through }}, {{ query }})
+    define_has_many_base_query({{ assoc_name }}, {{ model }}, {{ foreign_key }}, {{ through }}, {{ query }})
   end
 
-  private macro define_has_many_base_query(assoc_name, model, foreign_key, through)
+  private macro define_has_many_base_query(assoc_name, model, foreign_key, through, query)
     class BaseQuery
       def self.preload_{{ assoc_name }}(record)
         preload_{{ assoc_name }}(record: record, preload_query: {{ model }}::BaseQuery.new)
```

Comment:
```suggestion
        preload_{{ assoc_name }}(record: record, preload_query: {{ query }})
```

right? Since query could be nil, you might update the `has_many` macro to default it to `{{ model }}::BaseQuery.new` if missing

There's other places below that news up a query as well